server/auth/types/basic/routes.ts

```diff
@@ -171,7 +171,7 @@ export class BasicAuthRoutes {
             ? this.coreSetup.http.basePath.serverBasePath
             : '/';
           const requestQuery = request.url.query as ParsedUrlQueryParams;
-          if (requestQuery.nextUrl !== undefined) {
+          if (requestQuery?.nextUrl !== undefined) {
```

Comment:
I've created https://github.com/opensearch-project/security-dashboards-plugin/issues/936 to follow up, we should stop using the legacy url.query method